androguard/core/bytecodes/dvm_types.py

```diff
@@ -23,6 +23,43 @@ class TypeMapItem(IntEnum):
     ENCODED_ARRAY_ITEM = 0x2005
     ANNOTATIONS_DIRECTORY_ITEM = 0x2006
 
+    @staticmethod
+    def determine_load_order():
+        dependencies = {
+            TypeMapItem.HEADER_ITEM : set(),
+            TypeMapItem.STRING_ID_ITEM: set([TypeMapItem.STRING_DATA_ITEM]),
+            TypeMapItem.TYPE_ID_ITEM: set([TypeMapItem.STRING_ID_ITEM]),
+            TypeMapItem.PROTO_ID_ITEM: set([TypeMapItem.STRING_ID_ITEM, TypeMapItem.TYPE_ID_ITEM, TypeMapItem.TYPE_LIST]),
+            TypeMapItem.FIELD_ID_ITEM: set([TypeMapItem.STRING_ID_ITEM, TypeMapItem.TYPE_ID_ITEM]),
+            TypeMapItem.METHOD_ID_ITEM: set([TypeMapItem.STRING_ID_ITEM, TypeMapItem.TYPE_ID_ITEM, TypeMapItem.PROTO_ID_ITEM]),
+            TypeMapItem.CLASS_DEF_ITEM: set([TypeMapItem.TYPE_ID_ITEM, TypeMapItem.TYPE_LIST, TypeMapItem.STRING_ID_ITEM, TypeMapItem.DEBUG_INFO_ITEM, TypeMapItem.ANNOTATIONS_DIRECTORY_ITEM, TypeMapItem.CLASS_DATA_ITEM, TypeMapItem.ENCODED_ARRAY_ITEM]),
+            TypeMapItem.MAP_LIST: set(),
+            TypeMapItem.TYPE_LIST: set([TypeMapItem.TYPE_ID_ITEM]),
+            TypeMapItem.ANNOTATION_SET_REF_LIST: set([TypeMapItem.ANNOTATION_SET_ITEM]),
+            TypeMapItem.ANNOTATION_SET_ITEM: set([TypeMapItem.ANNOTATION_ITEM]),
+            TypeMapItem.CLASS_DATA_ITEM: set([TypeMapItem.FIELD_ID_ITEM, TypeMapItem.METHOD_ID_ITEM]),
+            TypeMapItem.CODE_ITEM: set([TypeMapItem.DEBUG_INFO_ITEM, TypeMapItem.TYPE_ID_ITEM]),
+            TypeMapItem.STRING_DATA_ITEM: set(),
+            TypeMapItem.DEBUG_INFO_ITEM: set([TypeMapItem.STRING_ID_ITEM, TypeMapItem.TYPE_ID_ITEM]),
+            TypeMapItem.ANNOTATION_ITEM: set([TypeMapItem.PROTO_ID_ITEM, TypeMapItem.STRING_ID_ITEM, TypeMapItem.TYPE_ID_ITEM, TypeMapItem.FIELD_ID_ITEM, TypeMapItem.METHOD_ID_ITEM]),
+            TypeMapItem.ENCODED_ARRAY_ITEM: set([TypeMapItem.PROTO_ID_ITEM, TypeMapItem.STRING_ID_ITEM, TypeMapItem.TYPE_ID_ITEM, TypeMapItem.FIELD_ID_ITEM, TypeMapItem.METHOD_ID_ITEM]),
+            TypeMapItem.ANNOTATIONS_DIRECTORY_ITEM: set([TypeMapItem.FIELD_ID_ITEM, TypeMapItem.METHOD_ID_ITEM, TypeMapItem.ANNOTATION_SET_ITEM])
+        }
+        ordered = dict()
```

Comment:
maybe this one needs an `OrderedDict` - dictionary keys are not ordered in python <3.6 - and at the moment we still support 3.4

Reply:
It doesn't need to be ordered because the result of the method is being sorted in MapList class